src/main/java/hudson/plugins/locale/LocaleFilter.java

```diff
@@ -29,21 +32,50 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             throws IOException, ServletException {
         if (request instanceof HttpServletRequest) {
             PluginImpl plugin = (PluginImpl) Jenkins.getActiveInstance().getPlugin("locale");
-            if (plugin != null && plugin.isIgnoreAcceptLanguage()) {
+            final Locale locale;
+            String currentUserLocale = getCurrentUserLocale();
+
+            if (plugin == null) {
+                chain.doFilter(request, response);
+                return;
+            } else if (plugin.isUserPrefer() && currentUserLocale != null) {
+                locale = PluginImpl.parse(currentUserLocale);

```

Comment:
It is likely to lock down the user if `IllegalArgumentException` is thrown. Would be better to suppress the exception. Bonus points fo caching the locale in the user property so that we do not have to parse it on every page refresh on the server side